pkg/util/hostport/hostport.go

```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package util
+package hostport
```

Comment:
I'm not a fan of this being in a directory referring to hostport. That makes it seems like its used only for hostPort when in reality you are using it for both hostPort and nodePort. I think a more appropriate place for this would be /pkg/util/net or pkg/util/port.

Reply:
I like pkg/util/net for this.

Like any good api extraction, it gives us a chance to examine the API and do better.  Comments below

Reply:
As general guidance - put the most important bits first.  In this case, I think the methods are most important.

What if we refactored this along the lines of:

```
// Track all ports in a PrtHolder
var currentPorts := utilnet.PortHolder{}

// Build up a new transaction for each sync loop.
txn := currentPorts.Replacement()
defer(txn.Abandon())  // in case we early-return

// Add each local port, even ones we already have.  If it is a transaction, it will copy from the parent.
txn.Open(ip, port, proto, desc)

// If we succeeded...
txn.Commit()  // will keep the new set of ports and close close ports we don't need

// If we are aborting the transaction, just return
```

Reply:
The nice thing here is we don't need to expose LocalPort or Closeable.  Nice and encapsulated.

Reply:
+100 to the above proposed refactor

Reply:
I just pushed an update and the basic idea is encapsulating `LocalPort` without exposing it. Sorry, I have tried but I just can't avoid expose `Closeable`.